components/kyma-operator/pkg/steps/installation.go

```diff
@@ -115,14 +120,30 @@ func (steps *InstallationSteps) processComponents(installationData *config.Insta
 
 		steps.PrintStep(stepName)
 
-		processErr := step.Run()
-		if steps.errorHandlers.CheckError("Step error: ", processErr) {
-			_ = steps.statusManager.Error(component.GetReleaseName(), stepName, processErr)
-			return processErr
+		err := retry.Do(
+			func() error {
+				processErr := step.Run()
+				if steps.errorHandlers.CheckError("Step error: ", processErr) {
+					_ = steps.statusManager.Error(component.GetReleaseName(), stepName, processErr)
+					return processErr
+				}
+				return nil
+			},
+			retry.Attempts(uint(len(steps.backoffIntervals))+1),
+			retry.DelayType(func(attempt uint, config *retry.Config) time.Duration {
+				log.Printf("Warning: Retry number %d (sleeping for %d[s]).\n", attempt+1, steps.backoffIntervals[attempt])
+				return time.Duration(steps.backoffIntervals[attempt]) * time.Second
+			}),
+		)
+		if err != nil {
+			err := steps.actionManager.RemoveActionLabel(installationData.Context.Name, installationData.Context.Namespace, "action")
```

Comment:
A hint: Duplicated code. I think it's enough to break out of the loop here. In order to avoid explicit `break` stmt, the body of the loop can be extracted to a separate function.

Reply:
I moved the code to the defer.

Reply:
update: I moved it to the function, that wraps an error and returns it